compiler/vm/vmdef.nim

```diff
@@ -155,27 +397,92 @@ type
   PStackFrame* = ref TStackFrame
   TStackFrame* {.acyclic.} = object
     prc*: PSym                 # current prc; proc that is evaluated
-    slots*: seq[TFullReg]      # parameters passed to the proc + locals;
+    slots*: SlotStorage      # parameters passed to the proc + locals;
                               # parameters come first
-    next*: PStackFrame         # for stacking
+    next*: int         # for stacking # TODO: probably not needed anymore now?
     comesFrom*: int
     safePoints*: seq[int]      # used for exception handling
                               # XXX 'break' should perform cleanup actions
                               # What does the C backend do for it?
   Profiler* = object
     tEnter*: float
-    tos*: PStackFrame
+    tos*: int   ## The current stack frame
 
   TPosition* = distinct int
 
   PEvalContext* = PCtx
 
+static:
+  doAssert sizeof(VmObjectAtom.atom) == sizeof(Atom)
```

Comment:
Put `atom: Atom` in there if you really need it to be like this.

Reply:
That's what I did in the beginning, but it caused issues due to `Atom` being a `{.union.}` that contains GC'ed types.

`VmObjectAtom` is queued for removal anyway and `Atom` will likely follow.